### PR TITLE
chore(functional-test):add functional test to the braze proxy repo

### PR DESCRIPTION
**Pull request by sri-kirsh**
## Goal
- add functional/e2e test so we can test the status code and response object returned by each endpoint.

### Implementation details/feedback-on
- for adding tests for other endpoints, any external calls  (e.g call to client-api) can be mocked (using nock) and we can test the response object and status code.

### CODE REVIEW

File: src/main.functional.ts [hunk 1]
```diff
@@ -0,0 +1,19 @@
+import { expect } from 'chai';
+import { app } from './main';
+
+describe('functional tests', () => {
+  /*eslint-disable */
+  //there are no types-supertest, so we use require.
+  //this throws lint error.
+  const request = require('supertest');
```
Comment (Herraj) [inline src/main.functional.ts:8]: We can't use the `import` syntax? e.g `import { request } from 'supertest' `

**Review by Herraj — APPROVED** [review]

File: src/main.functional.ts (hunk 1 above)
Comment (kschelonka) [reply src/main.functional.ts:8]: what about using nock?

File: src/main.functional.ts [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+  /*eslint-enable*/
+
+  describe('health endpoint', () => {
```
Comment (kschelonka) [inline src/main.functional.ts:11]: error callback test? can spy on sentry call

Comment (sri-kirsh) [reply src/main.functional.ts:11]: i am just looking to setup a skeleton for functional test, i will let product squad to continue with adding other test for their use cases.

File: src/main.functional.ts (hunk 1 above)
Comment (sri-kirsh) [reply src/main.functional.ts:8]: ya types/superagent might work with superagent, not supertest right? 
will change to either of this package and fix circle ci, and put for review again - changing PR to wip

Comment (sri-kirsh) [reply src/main.functional.ts]: found a @types/supertest, used it and we can import request now. https://www.npmjs.com/package/@types/supertest

**Review by kkyeboah — APPROVED** [review]

**Pull request merged**